ledger-core/virtual/integration/outgoing_test.go

```diff
@@ -0,0 +1,221 @@
+
+		callOutgoing = server.RandomLocalWithPulse()
+
+		outgoingCallRef = reference.NewRecordOf(objectAGlobal, callOutgoing)
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectAGlobal).
+			CallMethod(objectBGlobal, classB, "Bar", []byte("123")).
+			SetInterference(barIsolation.Interference).
+			SetIsolation(barIsolation.State)
+		objectBExecutionMock := runnerMock.AddExecutionMock("Foo")
+		objectBExecutionMock.AddStart(
+			func(ctx execution.Context) {
+				logger.Debug("ExecutionStart [A.Foo]")
+				assert.Equal(t, objectAGlobal, ctx.Object)
+				assert.Equal(t, callOutgoing, ctx.Request.CallOutgoing)
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Error:    nil,
+				Outgoing: outgoingCall,
+			},
+		)
+		objectBExecutionMock.AddContinue(
+			func(result []byte) {
+				logger.Debug("ExecutionContinue [A.Foo]")
+				require.Equal(t, []byte("finish B.Bar"), result)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish A.Foo"), objectAGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionClassify("Foo", fooIsolation, nil)
+	}
+
+	callCount := 0
+
+	firstPulse := server.GetPulse().PulseNumber
+
+	expectedToken := payload.CallDelegationToken{
+		TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+		PulseNumber:       server.GetPulse().PulseNumber,
+		Callee:            objectAGlobal,
+		Outgoing:          outgoingCallRef,
+		DelegateTo:        server.JetCoordinatorMock.Me(),
+	}
+
+	{
+		typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+			assert.Equal(t, firstPulse, report.OrderedPendingEarliestPulse)
+			assert.Equal(t, int32(1), report.OrderedPendingCount)
+			return false
+		})
+		typedChecker.VDelegatedCallRequest.Set(func(request *payload.VDelegatedCallRequest) bool {
+			assert.Zero(t, request.DelegationSpec)
+			assert.Equal(t, objectAGlobal, request.Callee)
+			assert.Equal(t, payload.BuildCallFlags(fooIsolation.Interference, fooIsolation.State), request.CallFlags)
+			assert.Equal(t, outgoingCallRef, request.CallOutgoing)
+
+			server.SendPayload(ctx, &payload.VDelegatedCallResponse{
+				Callee:                 request.Callee,
+				ResponseDelegationSpec: expectedToken,
+			})
+
+			return false
+		})
+		typedChecker.VDelegatedRequestFinished.Set(func(finished *payload.VDelegatedRequestFinished) bool {
+			assert.Equal(t, payload.CTMethod, finished.CallType)
+			assert.Equal(t, expectedToken, finished.DelegationSpec)
+			return false
+		})
+		typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+			assert.Equal(t, objectBGlobal, request.Callee)
+			assert.Equal(t, objectAGlobal, request.Caller)
+
+			assert.Equal(t, firstPulse, request.CallOutgoing.Pulse())
+			if callCount == 0 {
+				assert.Equal(t, payload.CallDelegationToken{}, request.DelegationSpec)
```

Comment:
you can use `request.DelegationSpec.IsZero()` (just proposal)

Reply:
when it fails - here it will tell you that expectations is empty CallDelegationToken, but received CallDelegationToken with values. assert.True will assert that wanted true, got false, which is less verbose

Reply:
well, on the second thought, I could switch to assert.Zero